### PR TITLE
Add Svelte unidriver

### PR DESCRIPTION
**Pull request by nktssh**

### CODE REVIEW

File: adapters/jsdom-svelte/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    isMultipleElementsWithLocatorError,
+    Locator,
+    MultipleElementsWithLocatorError,
+    NoElementWithLocatorError,
+    UniDriver, UniDriverList, waitFor
+} from "@unidriver/core";
+
+import {fireEvent} from '@testing-library/svelte';
+
+type HTMLFocusableElement = HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement | HTMLButtonElement;
```
Comment (giladsegal) [inline adapters/jsdom-svelte/src/index.ts]: `HTMLAnchorElement` is also focusable

File: adapters/jsdom-svelte/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+type ElementOrElementFinder = (() => Element) | Element | (() => Promise<Element>);
+type ElementsOrElementsFinder = (() => Element[]) | Element[] | (() => Promise<Element[]>);
+
+const isPromise = (a: Promise<any> | any ): a is Promise<any> => {
```
Comment (giladsegal) [inline adapters/jsdom-svelte/src/index.ts]:
```ts
Promise<any> | any
```
Is reduced to any

File: adapters/jsdom-svelte/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+                await fireEvent.mouseDown(el);
+            },
+            release: async () => {
+                const el = await elem();
+
+                await fireEvent.mouseUp(el);
+            },
+            moveTo: async (to) => {
+                const el = await elem();
+                const {left, top} = (await to.getNative()).getBoundingClientRect();
```
Comment (giladsegal) [inline adapters/jsdom-svelte/src/index.ts]: Will this work in all testing environments?

File: adapters/jsdom-react/package.json [hunk 4]
```diff
@@ -20,7 +20,7 @@
 		"@types/sinon": "^7.0.10",
 		"@unidriver/test-suite": "^1.1.5",
 		"chai": "^4.2.0",
-		"jsdom": "^15.1.1",
+		"jsdom": "15.1.1",
```
[the remark this answers is not in the document]
Comment (nktssh) [reply adapters/jsdom-react/package.json]: see https://github.com/jsdom/jsdom/issues/2586#issuecomment-564487539

File: adapters/jsdom-svelte/src/index.ts (hunk 1 above)
Comment (nktssh) [reply adapters/jsdom-svelte/src/index.ts]: worth to add to react also maybe?

File: adapters/jsdom-svelte/src/index.ts (hunk 3 above)
Comment (nktssh) [reply adapters/jsdom-svelte/src/index.ts]: Mostly this is re-implementation of the react driver, so if it works there it should work here as well. Not sure we tested all possible environments though

File: adapters/jsdom-svelte/src/index.ts (hunk 2 above)
Comment (nktssh) [reply adapters/jsdom-svelte/src/index.ts]: it's actually a type guard for TS

File: adapters/jsdom-svelte/src/index.ts (hunk 3 above)
Comment (GabiGrin) [reply adapters/jsdom-svelte/src/index.ts]: wdym by testing environment? jsdom / puppeteer/ etc?
it's only relevant to jsdom-svelte, on others you will use their own adapters

File: adapters/jsdom-svelte/src/index.ts (hunk 2 above)
Comment (giladsegal) [reply adapters/jsdom-svelte/src/index.ts]: I know but:
```ts
const isPromise = (a: Promise<any> | any ): a is Promise<any> => {
```
Is equivalent to 
```ts
const isPromise = (a: any ): a is Promise<any> => {
```

File: adapters/jsdom-react/package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (giladsegal) [reply adapters/jsdom-react/package.json]: I would add this line as a different commit with a comment to unlock this dependency once the issue is resolved

File: adapters/jsdom-svelte/src/index.ts (hunk 3 above)
Comment (giladsegal) [reply adapters/jsdom-svelte/src/index.ts]: Yeah you are right - sorry for that

File: adapters/jsdom-svelte/src/index.ts (hunk 1 above)
Comment (giladsegal) [reply adapters/jsdom-svelte/src/index.ts]: Lets ask @GabiGrin, changing react adapter might break existing tests

Comment (GabiGrin) [reply adapters/jsdom-svelte/src/index.ts]: @nktssh  @giladsegal but then it means it _should_ break, no? 🤔

**Comment by GabiGrin** [comment]
@giladsegal  @hadarge  @nktssh @NickVlasov LGTM, can merge & release?

**Comment by NickVlasov** [comment]
LGTM also

**Comment by nktssh** [comment]
Looks good to me 🙂

**Comment by hadarge** [comment]
Yes, please merge, release and ping me so I can use it 😬

**Comment by GabiGrin** [comment]
Screen Shot 2019-12-17 at 18 00 38

**Comment by shlomitc** [comment]
Amazing job everybody! 👏

**Pull request merged**